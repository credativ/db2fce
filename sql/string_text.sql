SET search_path = db2,pg_catalog,public;

-- LOCATE()
SELECT LOCATE('N', 'DINING') FROM SYSIBM.SYSDUMMY1;

-- TRANSLATE()
SELECT TRANSLATE('12345', 'ax', '143') FROM SYSIBM.SYSDUMMY1;
SELECT TRANSLATE('Hanauma Bay') FROM SYSIBM.SYSDUMMY1;
SELECT TRANSLATE('Hanauma Bay', 'j', 'B'	) FROM SYSIBM.SYSDUMMY1;
SELECT TRANSLATE('Hanauma Bay', 'ei', 'aa') FROM SYSIBM.SYSDUMMY1;
SELECT TRANSLATE('Hanauma Bay', 'r', 'Bu') FROM SYSIBM.SYSDUMMY1;

-- HEX()
SELECT HEX('000020') FROM SYSIBM.SYSDUMMY1;
SELECT HEX(000020) FROM SYSIBM.SYSDUMMY1; -- WRONG

-- UPPER(text, text)
SELECT UPPER('Müller', 'de_DE') FROM SYSIBM.SYSDUMMY1;

-- LOWER(text, text)
SELECT LOWER('MÜLLER', 'de_DE') FROM SYSIBM.SYSDUMMY1;
