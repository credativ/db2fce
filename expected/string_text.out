SET search_path = db2,pg_catalog,public;
-- LOCATE()
SELECT LOCATE('N', 'DINING') FROM SYSIBM.SYSDUMMY1;
 locate 
--------
      3
(1 row)

-- TRANSLATE()
SELECT TRANSLATE('12345', 'ax', '143') FROM SYSIBM.SYSDUMMY1;
 translate 
-----------
 a2x5
(1 row)

SELECT TRANSLATE('Hanauma Bay') FROM SYSIBM.SYSDUMMY1;
  translate  
-------------
 HANAUMA BAY
(1 row)

SELECT TRANSLATE('Hanauma Bay', 'j', 'B'	) FROM SYSIBM.SYSDUMMY1;
  translate  
-------------
 Hanauma jay
(1 row)

SELECT TRANSLATE('Hanauma Bay', 'ei', 'aa') FROM SYSIBM.SYSDUMMY1;
  translate  
-------------
 Heneume Bey
(1 row)

SELECT TRANSLATE('Hanauma Bay', 'r', 'Bu') FROM SYSIBM.SYSDUMMY1;
 translate  
------------
 Hanama ray
(1 row)

-- HEX()
SELECT HEX('000020') FROM SYSIBM.SYSDUMMY1;
     hex      
--------------
 303030303230
(1 row)

SELECT HEX(000020) FROM SYSIBM.SYSDUMMY1; -- WRONG
ERROR:  function hex(integer) does not exist
LINE 1: SELECT HEX(000020) FROM SYSIBM.SYSDUMMY1;
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

-- UPPER(text, text)
SELECT UPPER('Müller', 'de_DE') FROM SYSIBM.SYSDUMMY1;
 upper
--------
 MÜLLER
(1 row)

-- LOWER(text, text)
SELECT LOWER('MÜLLER', 'de_DE') FROM SYSIBM.SYSDUMMY1;
 lower  
--------
 müller
(1 Zeile)
